src/plot_api/plot_api.js

```diff
@@ -1686,6 +1686,74 @@ function _restyle(gd, aobj, traces) {
     };
 }
 
+/**
+ * Converts deprecated layout attributes to the current API
+ * to ensure backwards compatibility.
+ *
+ * @param layoutObj
+ */
+function cleanDeprecatedLayoutAttributes(layoutObj) {
+
+    // Support old-style update of the title's font
+    var isTitlefontSet = layoutObj.titlefont;
+    var isFontInTitleSet = layoutObj.title && layoutObj.title.font;
+    var isFontInTitleNotSet = !isFontInTitleSet;
+    if(isTitlefontSet && isFontInTitleNotSet) {
+
+        // Use string attribute because initiating a new title object
+        // to be able to specify a font property on it would require to
+        // know the potentially existing `title.text` property.
+        layoutObj['title.font'] = layoutObj.titlefont;
+        delete layoutObj.titlefont;
+    }
+
+    // Note, that updating x-axis and y-axis title fonts
+    // was never supported unless (i) using string
+    // attributes or (ii) passing `xaxis.title` / `yaxis.title`
+    // again. And these cases are covered by
+    // helpers.cleanLayout anyways.
+}
+
+/**
+ * Converts deprecated layout "string attributes" to
+ * "string attributes" of the current API to ensure backwards compatibility.
+ *
+ * E.g. Maps {'xaxis.title': 'A chart'} to {'xaxis.title.text': 'A chart'}
+ *
+ * @param layoutObj
+ */
+function cleanDeprecatedLayoutAttributeStrings(layoutObj) {
+    var oldAxisTitleRegExp = /axis\d{0,2}.title$/;
```

Comment:
We should reuse 

https://github.com/plotly/plotly.js/blob/11bec79d010a44694bf70b357bd11e11ae874ccb/src/plots/cartesian/constants.js#L20

:recycle:

Reply:
Ha I see, this is use for all (cartesian, ternary, polar, scene) axes, so we can't just reuse the cartesian `attrRegex`. We should create a new regex: `counterRegex('axis')` instead.

Reply:
I have considered using it. Problem right now is that all regular expressions start with `^` at the moment. 

Should I add another optional parameter to `counter` that allows to opt out of that?

Reply:
Ha good point. 

> Should I add another optional parameter to counter that allows to opt out of that?

That, or just replacing `\d{0,2}` with `([2-9]|[1-9][0-9]+)?` to match the same axis name patterns. Your call.

Reply:
Okay. I've added another parameter to `Lib.counterRegex` and reused it here.